File: Binary Search Tree/110-Balanced Binary Tree.py
class Solution:
    def isBalanced(self, root: Optional[TreeNode]) -> bool:
        
        def isheight(root):
            if root==None:
                return 0
            l=isheight(root.left)
            r=isheight(root.right)
            
            if l==-1 or r==-1:
                return -1
            
            if abs(l-r)>1:
                return -1
            return max(l,r)+1
        
        return isheight(root) != -1

File: Binary Search Tree/Java/110. Balanced Binary Tree.java
*******************************Check if Balance Binary Tree*****************************************
class Solution {
    public boolean isBalanced(TreeNode root) {
        return dfsHeight(root)!=-1;
    }
    int dfsHeight(TreeNode root)
    {
        if(root==null) return 0;
        int lh=dfsHeight(root.left);
        if(lh==-1) return -1;
        int rh=dfsHeight(root.right);
        if(rh==-1) return -1;
        if(Math.abs(lh-rh)>1) return -1;

        //calculate height
        return 1+Math.max(lh,rh);
    }
}


